ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/components/ScanConfigurationSelect.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { useContext, useState } from 'react';
+import {
+    Divider,
+    MenuToggle,
+    MenuToggleElement,
+    Select,
+    SelectGroup,
+    SelectList,
+    SelectOption,
+    Spinner,
+} from '@patternfly/react-core';
+
+import { ScanConfigurationsContext } from '../ScanConfigurationsProvider';
+
+const ALL_SCAN_SCHEDULES_OPTION = 'All scan schedules';
+
+function ScanConfigurationSelect() {
+    const [isOpen, setIsOpen] = useState(false);
+
+    const { scanConfigurationsQuery, selectedScanConfig, setSelectedScanConfig } =
+        useContext(ScanConfigurationsContext);
+
+    const onToggleClick = () => {
+        setIsOpen(!isOpen);
+    };
+
+    const onSelect = (
+        _event: React.MouseEvent<Element, MouseEvent> | undefined,
+        value: string | number | undefined
+    ) => {
+        const selectedValue = value === ALL_SCAN_SCHEDULES_OPTION ? undefined : (value as string);
+        setSelectedScanConfig(selectedValue);
+        setIsOpen(false);
+    };
+
+    const renderToggle = (toggleRef: React.Ref<HTMLButtonElement | MenuToggleElement>) => {
+        return (
+            <MenuToggle ref={toggleRef} onClick={onToggleClick} isExpanded={isOpen}>
+                {(selectedScanConfig as string) || ALL_SCAN_SCHEDULES_OPTION}
+            </MenuToggle>
+        );
+    };
+
+    return (
+        <Select
+            id="scan-schedules-filter-id"
+            isOpen={isOpen}
+            selected={selectedScanConfig || ALL_SCAN_SCHEDULES_OPTION}
+            onSelect={onSelect}
+            onOpenChange={(isOpen) => setIsOpen(isOpen)}
+            toggle={renderToggle}
+            shouldFocusToggleOnSelect
+        >
+            {scanConfigurationsQuery.isLoading ? (
+                <SelectOption isLoading value="loader">
+                    <Spinner size="lg" />
+                </SelectOption>
+            ) : (
+                <>
+                    <SelectGroup label="View all results">
+                        <SelectList>
+                            <SelectOption
+                                key="key_all-scan-schedules"
```

Comment:
For info, not a request for change: if I understand correctly, React expects `key` prop:
* for `map` method
* for array of elements
but not for inline rendering nor even for conditional rendering with `&&` idiom.

On the other side of the coin, in rare cases, event handlers might change the value of a `key` prop not required by React to delete an instance and create a new instance of the element.